sdk/core/Azure.Core/src/Shared/AsyncLockWithValue.cs

```diff
@@ -1,25 +1,63 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#nullable enable
+
 using System;
 using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
 using System.Threading;
 using System.Threading.Tasks;
 using Azure.Core.Pipeline;
 
-namespace Azure.Identity
+namespace Azure.Core
 {
     /// <summary>
     /// Primitive that combines async lock and value cache
     /// </summary>
     /// <typeparam name="T"></typeparam>
     internal sealed class AsyncLockWithValue<T>
     {
-        private readonly object _syncObj = new object();
-        private Queue<TaskCompletionSource<Lock>> _waiters;
+        private readonly object _syncObj = new();
+        private Queue<TaskCompletionSource<LockOrValue>>? _waiters;
```

Comment:
How are we regression testing any libraries that use this type?

Reply:
There is no change in functionality. It is just that `Azure.Identity` doesn't have nullable reference types analysis enabled, so it didn't require `_waiters` field to be marked as "can be null".